packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -860,11 +860,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.blobManager = new BlobManager(
             this.IFluidHandleContext,
             () => {
-                assert(this.attachState !== AttachState.Detached, 0x123 /* "Blobs NYI in detached container mode" */);
                 return this.storage;
             },
-            (blobId) => this.submit(ContainerMessageType.BlobAttach, undefined, undefined, { blobId }),
-            (fn) => this.once("dispose", fn),
+            (blobId) => {
+                if (this.attachState === AttachState.Attached) {
```

Comment:
We need to decide what is the semantics (in the moment) for blob upload while in Attaching state. I'd add assert here as well that it's not supported.
The alternative is to implement upload by waiting for file creation (attachment) to finish before continuing upload.